dlrs/main/classes/ManageLookupRollupSummariesNewController.cls

```diff
@@ -0,0 +1,364 @@
+            if(selectedChildObject==null &&   errormessage.contains('Attempt to de-reference a null object')){
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, 'Please Select the Child Object'));  
+            }else if(selectedRField==null && errormessage.contains('Attempt to de-reference a null object')){
+                ApexPages.addMessage(new ApexPages.Message(ApexPages.Severity.Error, 'Please Select the Relationship Field')); 
+            }else{
+                ApexPages.addMessages(e);
+            }
+        }
+        return null;
+    }
+    
+    public PageReference deleteX() {
+        /* AJB 16/11/21: whilst this is referenced from the page, the button's only visible if there's a QSP with an Id, i.e. edit mode, which isn't supported in this version
+        try {
+            // Delete the rollup custom metadata
+            RollupService.delete_X(new List<String> { LookupRollupSummary.DeveloperName });
+            // Reload this page (and thus the rollup list in a new request, metadata changes are not visible until this request ends)
+            PageReference newPage = Page.managelookuprollupsummaries;
+            newPage.setRedirect(true);
+            return newPage;                 
+        } catch (Exception e) {
+            ApexPages.addMessages(e);
+        }*/
+        return null;
+    }
+
+    public PageReference hideMessage() {
+        DeclarativeLookupRollupSummaries__c settings = DeclarativeLookupRollupSummaries__c.getOrgDefaults();
+        settings.HideManageLookupRollupSummariesInfo__c = true;
+        upsert settings;
+        return null;
+    }
+
+    public List<SelectOption> getCalculationModes(){
+        List<SelectOption> options = new List<SelectOption>();
+        for(Schema.Picklistentry picklistEntry : LookupRollupSummary__c.fields.CalculationMode__c.getDescribe().getPickListValues()) {
+            options.add(new SelectOption(picklistEntry.getValue(), picklistEntry.getLabel()));
+        }
+        return options;
+    }           
+
+    public List<SelectOption> getAggregateOperations(){
+        List<SelectOption> options = new List<SelectOption>();
+        for(Schema.Picklistentry picklistEntry : LookupRollupSummary__c.fields.AggregateOperation__c.getDescribe().getPickListValues()) {
+            options.add(new SelectOption(picklistEntry.getValue(), picklistEntry.getLabel()));
+        }
+        return options;
+    }           
+
+    public List<SelectOption> getCalculationSharingModes(){
+        List<SelectOption> options = new List<SelectOption>();
+        for(Schema.Picklistentry picklistEntry : LookupRollupSummary__c.fields.CalculationSharingMode__c.getDescribe().getPickListValues()) {
+            options.add(new SelectOption(picklistEntry.getValue(), picklistEntry.getLabel()));
+        }
+        return options;
+    }
+
+    public List<Selectoption> getParentObjList(){
+        objectDescription = Schema.getGlobalDescribe();
```

Comment:
I think we talked about the fact that this can take a long time and return a huuuuuuge list